src/types/Patterns.ts

```diff
@@ -18,9 +18,22 @@
  * SPDX-License-Identifier: Apache-2.0
  ********************************************************************************/
 
+import { getRequireHttpsUrlPattern } from 'services/EnvironmentService'
+
+// check the REQUIRE_HTTPS_URL_PATTERN environment variable, defaulting to 'true' if not set
+const requireHttpsUrlPattern = getRequireHttpsUrlPattern() === 'true'
```

Comment:
the comment and the logic don't match - with the check 
`getRequireHttpsUrlPattern() === 'true'`
the default is http (example 'foo' === 'true' = false) so I suggest to reverse the check to 
`getRequireHttpsUrlPattern() !== 'false'`
then it will be http only if the value is 'false' and https for everything else

Reply:
done